src/hotspot/os_cpu/linux_aarch64/atomic_linux_aarch64.S

```diff
@@ -26,67 +26,121 @@
         .globl aarch64_atomic_fetch_add_8_default_impl
         .align 5
 aarch64_atomic_fetch_add_8_default_impl:
+        prfm    pstl1strm, [x0]
 0:      ldaxr   x2, [x0]
         add     x8, x2, x1
         stlxr   w9, x8, [x0]
         cbnz    w9, 0b
+        dmb     ish
         mov     x0, x2
         ret
 
         .globl aarch64_atomic_fetch_add_4_default_impl
         .align 5
 aarch64_atomic_fetch_add_4_default_impl:
+        prfm    pstl1strm, [x0]
 0:      ldaxr   w2, [x0]
         add     w8, w2, w1
         stlxr   w9, w8, [x0]
         cbnz    w9, 0b
+        dmb     ish
         mov     w0, w2
         ret
 
         .globl aarch64_atomic_xchg_4_default_impl
         .align 5
 aarch64_atomic_xchg_4_default_impl:
+        prfm    pstl1strm, [x0]
 0:      ldaxr   w2, [x0]
         stlxr   w8, w1, [x0]
         cbnz    w8, 0b
+        dmb     ish
         mov     w0, w2
         ret
 
         .globl aarch64_atomic_xchg_8_default_impl
         .align 5
 aarch64_atomic_xchg_8_default_impl:
+        prfm    pstl1strm, [x0]
 0:      ldaxr   x2, [x0]
         stlxr   w8, x1, [x0]
         cbnz    w8, 0b
+        dmb     ish
         mov     x0, x2
         ret
 
         .globl aarch64_atomic_cmpxchg_1_default_impl
         .align 5
 aarch64_atomic_cmpxchg_1_default_impl:
+        dmb     ish
```

Comment:
Having argued above that this dmb is never needed why is it in this default impl? (also for size 4 and 8)

Reply:
The default impl uses  ARMv8 LDXR ... STXR but the DMB is not needed if and only if ARMv8.1 LSE instructions.

Reply:
Doh! And it says that clearly in the comment. Ok.